### ISSUE TITLE
Better identify the category of error messages (e.g. "CSS parsing error")

### ISSUE THREAD

**Issue by rdeltour** [w3c/epubcheck#302]
_From [stickyje...@gmail.com](https://code.google.com/u/106880803105210833503/) on September 10, 2013 03:33:43_

PROBLEM: The attached invalid_media_type.epub file fails epubcheck.  Specifically, in its content.opf file, the <item> definitions for the .otf fonts have the incorrect media-type.  They should be "application/x-font-opentype", but are mistakenly defined as "text/css".  As otf files are binary, interpreting them as text/css generates hundreds of "is not allowed here" warnings.  For example:

```
WARNING: invalid_media_type.epub/OEBPS/Fonts/DejaVuSans-Bold.otf(5): The character 'NEWLINE' is not allowed in 'STRING' expressions
WARNING: invalid_media_type.epub/OEBPS/Fonts/LinLibertine_Re-4.1.8.otf(5145): Token '@????' not allowed here
```

There is no clear indication that something is wrong with the media-type.

METHOD:
(1) To reproduce the problem, run epubcheck on invalid_media_type.epub.  You will get a long list of warnings
(2) I also an attached correct_media_type.epub, where the media-type values in content.opf have been corrected. 

RESULT: A long list of "is not allowed here" warnings.

EXPECTED: It would be ideal if there was an error (or warning) indicating the exact source of the problem.  Although we have no interest in using it, the FlightCrew utility does give clear errors, including the filename and the line number, for example:

```
invalid_media_type.epub/OEBPS/content.opf    19    The <item> element's "media-type" attribute has value "text/css", but the file's media type is "application/vnd.ms-opentype".
```

A message similar to that would be perfect.

Even a more general message would be helpful, something like:  "In content.opf, an <item> has a media-type declared as a text mimetype, but is referencing a binary file."

PLATFORM: I am using epubcheck-3.0.1, on both OSX 10.8.5 (dev) and Linux 2.6.18-194.el5 (prod)

**Attachment:** [invalid_media_type.epub correct_media_type.epub](http://code.google.com/p/epubcheck/issues/detail?id=303)

_Original issue: http://code.google.com/p/epubcheck/issues/detail?id=303_

**Comment by rdeltour** [comment]
Doing type sniffing is out of scope for EPUBCheck, but I suppose we could better identified these messages are related to CSS parsing, which could already give a hint of what's happening.

Let's try to tackle that during the message refactoring after v4.2.0.
